### PR TITLE
Add device emulation

### PR DESCRIPTION
**Pull request by nekuz0r**
Add two methods to webContents to enable/disable device emulation.

### CODE REVIEW

**Comment by zcbenz** [comment]
This is awesome! Can you also add some documents for them in `docs/api/web-contents.md`?

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -640,6 +643,76 @@ bool WebContents::IsDevToolsOpened() {
...
+
+  blink::WebDeviceEmulationParams params;
+
+  if (dict.HasKey("screenPosition")) {
+    std::string screenPosition;
+    if (!dict.GetString("screenPosition", &screenPosition))
+      return;
+
+    // lower case
+    std::transform(screenPosition.begin(), screenPosition.end(),
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: We usually use `base::StringToLowerASCII` for this purpose.

File: atom/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -640,6 +643,76 @@ bool WebContents::IsDevToolsOpened() {
...
 }
 
+void WebContents::EnableDeviceEmulation(const base::DictionaryValue& dict) {
+  if (type_ == REMOTE)
+    return;
+
+  blink::WebDeviceEmulationParams params;
+
+  if (dict.HasKey("screenPosition")) {
+    std::string screenPosition;
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: In C++ we use `screen_position` instead `screenPosition` for variable names, we follows Google C++ style guide in Electron and Chromium, the code in `blink` follows a different style guide though.

File: atom/browser/api/atom_api_web_contents.cc [hunk 3]
```diff
@@ -640,6 +643,76 @@ bool WebContents::IsDevToolsOpened() {
...
+    if (!dict.GetString("screenPosition", &screenPosition))
+      return;
+
+    // lower case
+    std::transform(screenPosition.begin(), screenPosition.end(),
+                   screenPosition.begin(), ::tolower);
+
+    if (screenPosition.compare("mobile") == 0) {
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: We usually do string comparison with `screenPosition == "mobile"` in C++.

**Comment by nekuz0r** [comment]
@zcbenz Added documentation and updated code based on your remarks.

**Comment by zcbenz** [comment]
Awesome, thanks!

**Pull request merged**